pkg/kubelet/cm/cpumanager/cpu_manager.go

```diff
@@ -493,3 +512,19 @@ func (m *manager) updateContainerCPUSet(containerID string, cpus cpuset.CPUSet)
 func (m *manager) GetCPUs(podUID, containerName string) cpuset.CPUSet {
 	return m.state.GetCPUSetOrDefault(podUID, containerName)
 }
+
+func (m *manager) updateAdmittedPods(pod *v1.Pod) {
+	m.Lock()
+	defer m.Unlock()
+
+	// the previous pods passed the admit phase and we can remove the pod from the admitted pods
+	for podUID := range m.admittedPods {
+		if podUID != string(pod.UID) {
+			delete(m.admittedPods, podUID)
+		}
+	}
+
+	if _, ok := m.admittedPods[string(pod.UID)]; !ok {
+		m.admittedPods[string(pod.UID)] = pod
+	}
+}
```

Comment:
I don't quite understand why we need all of the extra logic in this function.

It currently says:
1) If the pod being considered is not in the `admittedPods` map, then remove it
2) If the pod being considered is not yet in the `admittedPods` map, put it in it

Which means that at any given time, only one pod should ever be present in the `admittedPods` map.

As such, we can just use a single variable (instead of a map) to hold the "currently being admitted pod", reducing the logic to:
```
func (m *manager) updatePodPendingAdmission(pod *v1.Pod) {
	m.Lock()
	defer m.Unlock()
	m.pendingAdmissionPod = pod
```

Reply:
True, I just somehow was affected by the re-use memory variable.

Reply:
> Which means that at any given time, only one pod should ever be present in the admittedPods map.
I thought the same, but IIUC (and @cynepco3hahue please correct me if I'm wrong) this holds true iff we can remove the pod from the internal map once the admission is complete. The current code (hence the IIUC) seems to prefer to keep the pod lingering a bit while, and perform the cleanup once a new pod gets admitted. If I got it right I don't have strong feelings about this approach, besides wasting a bit of memory it should be safe.

Reply:
My approach also leaves it "lingering" -- you only overwrite the variable the next time this function is called (which is at the admission of the next pod).

The main issue I see with the `map` approach is that we will also need to add a "getter" function since the `reconcileState()` function runs asynchronous to this one. Otherwise, we will run into concurrent map access panics.